### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by leeanndrees**
The app will now display error messages to the user if something goes wrong in getting their location or fetching weather data.

### CODE REVIEW

File: dressForTheWeather/Networking/WeatherNetworking.swift [hunk 1]
```diff
@@ -16,9 +16,12 @@ final class WeatherNetworking {
     static let weatherSession = URLSession(configuration: .default)
     static var dataTask: URLSessionDataTask?
     
+    static let darkSkyKey = "e3c6c3f5254ee83b29cf829a4eee5c46"
```
Comment (levosins) [inline dressForTheWeather/Networking/WeatherNetworking.swift]: ❓ should probably remove this key?

File: dressForTheWeather/Networking/WeatherNetworking.swift [hunk 2]
```diff
@@ -31,10 +34,11 @@ final class WeatherNetworking {
             
             if let error = error {
                 print(error.localizedDescription)
```
Comment (levosins) [inline dressForTheWeather/Networking/WeatherNetworking.swift]: You could probably remove this print call

File: dressForTheWeather/Networking/WeatherNetworking.swift [hunk 3]
```diff
@@ -16,9 +16,12 @@ final class WeatherNetworking {
     static let weatherSession = URLSession(configuration: .default)
     static var dataTask: URLSessionDataTask?
     
+    static let darkSkyKey = "e3c6c3f5254ee83b29cf829a4eee5c46"
+    
     // MARK: - Methods
     
-    static func getWeatherFor(latitude: Double, longitude: Double, completion: (@escaping (WeatherData) -> Void)) {
+    static func getWeatherFor(latitude: Double, longitude: Double, success: @escaping (WeatherData) -> Void, failure: @escaping (Error?) -> Void) {
```
Comment (levosins) [inline dressForTheWeather/Networking/WeatherNetworking.swift]: The Error parameter shouldn't be optional now that we have a closure that will specifically be called if there's an error.

File: dressForTheWeather/View Models/RecommendationViewModel.swift [hunk 4]
```diff
@@ -60,10 +61,14 @@ final class RecommendationViewModel {
...
     
     private func getTemperature(latitude: Double, longitude: Double) {
-        WeatherNetworking.getWeatherFor(latitude: latitude, longitude: longitude) { weatherData in
+        WeatherNetworking.getWeatherFor(latitude: latitude, longitude: longitude, success: {
+            weatherData in
             self.temperature = weatherData.currently.temperature
             self.setRecommendations(for: self.temperature)
-        }
+        }, failure: { error in
+            self.delegate?.didFailToGetWeather(error?.localizedDescription ?? "Something went wrong")
```
Comment (levosins) [inline dressForTheWeather/View Models/RecommendationViewModel.swift]: once the Error parameter isn't optional you won't need the ?? fallback

**Review by levosins — CHANGES_REQUESTED** [review]
looks good i just had some small changes

**Comment by leeanndrees** [comment]
@levosins I implemented you suggestions - ready for another look when you have a moment

**Pull request merged**